src/oic/oic/provider.py

```diff
@@ -1654,6 +1662,24 @@ def read_registration(self, authn, request, **kwargs):
         return Response(response.to_json(), content="application/json",
                         headers=[("Cache-Control", "no-store")])
 
+    def alter_registration(self, authn, request, **kwargs):
+        """Method to alter the client info on server side.
+
+        :param authn: Authorization HTTP header
+        :param request: Query part of the request
+        :return: Response with updated client info
+        """
+        return error_response('Unsupported operation', descr='Altering of the registration is not supported')
```

Comment:
I would suggest a 403 response, in line with the RFC 7592 2.2 section, as that spec is likely the one someone would implement here. It demands a 403.